### ISSUE TITLE
Regression? Refactor fields: OGR driver for '' not found

### ISSUE THREAD

**Issue by BergWerkGIS** [qgis/QGIS#40160]
**Describe the bug**

QGIS (OGR?) fails to determine output file type (gpkg in my case) when there are several dots in the output file name.
Worked with previous versions but doesn't seem to work with 3.16.0

**How to Reproduce**

Using output file name `theoutput-v100.gpkg` works:

![image](https://user-images.githubusercontent.com/4549888/99526662-d8dd8d00-299b-11eb-91ed-61540b9ad403.png)

Using output file name `theoutput-v1.0.0.gpkg` does not:

![image](https://user-images.githubusercontent.com/4549888/99526841-28bc5400-299c-11eb-97e1-b33aa465c8f4.png)

**QGIS and OS versions**

. | .
---- | ----
QGIS version  | 3.16.0-Hannover
QGIS code revision | 43b64b13f3
Compiled against | Qt 5.12.8
Running against | Qt 5.12.8
Compiled against | GDAL/OGR 3.0.4
Running against | GDAL/OGR | 3.0.4
Compiled against  | GEOS 3.8.0-CAPI-1.13.1
Running against | GEOS 3.8.0-CAPI-1.13.1 
Compiled against | SQLite 3.31.1
Running against | SQLite 3.31.1
PostgreSQL Client Version | 12.4 (Ubuntu 12.4-0ubuntu0.20.04.1)
SpatiaLite Version | 4.3.0a
QWT Version | 6.1.4
QScintilla2 Version | 2.11.2
Compiled against | PROJ 6.3.1
Running against | PROJ Rel. 6.3.1, February 10th, 2020
OS Version | Ubuntu 20.04.1 LTS
Active python plugins  | firstaid; pluginbuilder3; plugin_reloader; db_manager; processing

**Additional context**

**Comment by gioman** [comment]
>  Worked with previous versions but doesn't seem to work with 3.16.0

@BergWerkGIS it happens only in the modeller or also if you you the standalone tool?

**Comment by BergWerkGIS** [comment]
@gioman standalone tool as well.

![image](https://user-images.githubusercontent.com/4549888/99528009-f27fd400-299d-11eb-9990-4e26631227b1.png)

**Comment by BergWerkGIS** [comment]
@gioman my use case is a rather large model that has been exported to Python and integrated into a complex workflow driven by a plugin.
Are you by chance aware of any additional parameters I could pass in when defining the output for `Refactor fields` to force GPKG?

That's how I do it currently:
```python
'OUTPUT': 'ogr:dbname=\'' + export_path + '/' + export_base_name + '.gpkg\' table=\"current\" (geometry)'
```

**Comment by agiudiceandrea** [comment]
It seem the issue is in
QgsProcessingUtils::parseDestinationString
https://github.com/qgis/QGIS/blob/fb48bc92e662c31b12686862526400073d5565b2/src/core/processing/qgsprocessingutils.cpp#L672-L674
where the extension is calculated using completeSuffix(), i.e. everything after the FIRST point, so it is not correctly recognized for fi.le.name.ext e considered invalid
https://github.com/qgis/QGIS/blob/fb48bc92e662c31b12686862526400073d5565b2/src/core/processing/qgsprocessingprovider.cpp#L170-L175

This is similar to https://github.com/qgis/QGIS/issues/36353 (see also https://github.com/qgis/QGIS/pull/36371/files) about raster layer output.

@BergWerkGIS, maybe you could remove the dot from the filename or you could simply use `'OUTPUT': 'drive:/path/theoutput-v1.0.0.gpkg'` (in this case you cannot directly control the layer name).

**Comment by agiudiceandrea** [comment]
The issue didn't happen with 3.4.5. It happens with 3.4,9 and later.

**Comment by BergWerkGIS** [comment]
@agiudiceandrea thanks for your suggestion but I need full control over file name, layer name and name of the geometry column as the refactor-fields step is embedded in a larger workflow (outside of QGIS).
My workaround for now is to use a file name with just one dot when calling `processing.run('qgis.refactorfields')` and rename the result after the processing step is done.

**Comment by roya0045** [comment]
@agiudiceandrea in this case using suffix() instead of completeSuffix() would solve this issue. Unsure if this would also have side effects.

**Comment by agiudiceandrea** [comment]
Yes, indeed. Maybe @m-kuhn and @volaya could know a little more about it (229c617785198aa32aac2104e8f842e4b35a39ab, 13bff9620c39e110852f4623e3ad9c3ec5ba260c). See also https://github.com/qgis/QGIS/issues/36353#issuecomment-626809574.

**Comment by nyalldawson** [comment]
You'd just need to do a bit of a study into the gdal drivers to see whether any formats use . in their extensions first. (@rouault are you aware of any?) If not, should be a safe change to make!

**Comment by roya0045** [comment]
@nyalldawson I assume this would be to ensure that filenames with . don't get converted to extensions if its the uri containing the layer string after the database name.

**Comment by Pedro-Murteira** [comment]
@wilhelmberg Hello, is this issue still valid on recent releases?

**Comment by wilhelmberg** [comment]
@Pedro-Murteira sorry, but I can't say. I'm not part of that particular project anymore.

**Comment by mblesius** [comment]
> @wilhelmberg Hello, is this issue still valid on recent releases?

Yes, the issue is still valid on QGIS 3.30.2 's-Hertogenbosch
